test/query.test.js

```diff
@@ -4455,4 +4455,342 @@ describe('Query', function() {
       assert.strictEqual(target, null);
     });
   });
+
+
+  describe('Query with requireFilter', function() {
+    let Person;
+    let _id;
+  
+    beforeEach(async function() {
+      this.timeout(15000);
+  
+      try {
+       
+        if (mongoose.connection.readyState !== 1) {
+          await mongoose.connect('mongodb://localhost:27017/testdb');
```

Comment:
Don't call `mongoose.connect()` in tests. We use `db.model()` in tests throughout.